src/components/Quiz/QuizQuestion.tsx

```diff
@@ -0,0 +1,104 @@
+// Libraries
+import React, { useMemo } from "react"
+import { Box, Circle, Text, useColorMode } from "@chakra-ui/react"
+
+// Components
+import Button from "../Button"
+
+// Types
+import { Question } from "../../types"
+
+export interface IProps {
+  questionData: Question
+  showAnswer: boolean
+  handleSelection: (answerId: string) => void
+  selectedAnswer: string | null
+}
+
+const QuizQuestion: React.FC<IProps> = ({
+  questionData,
+  showAnswer,
+  handleSelection,
+  selectedAnswer,
+}) => {
+  const { colorMode } = useColorMode()
+  const isDarkMode = colorMode === "dark"
+  const { answers, prompt, correctAnswerId } = questionData
+
+  // Memoized values
+  const explanation = useMemo<string>(() => {
+    if (!selectedAnswer) return ""
+    return answers.filter(({ id }) => id === selectedAnswer)[0].explanation
+  }, [selectedAnswer])
+  const buttonBg = useMemo<string>(() => {
+    if (!showAnswer) return "primary"
+    return correctAnswerId === selectedAnswer ? "green.500" : "red.500"
+  }, [questionData, selectedAnswer, showAnswer])
+  const letterColor = useMemo<string>(() => {
+    if (!showAnswer) return "white"
+    return correctAnswerId === selectedAnswer ? "green.500" : "red.500"
+  }, [questionData, selectedAnswer, showAnswer])
+  return (
+    <Box w={"100%"}>
+      <Text fontWeight={"700"} fontSize={"2xl"}>
+        {prompt}
+      </Text>
+      {answers.map(({ id, label }, index) => {
+        const active = selectedAnswer === id
+        const iconBackgroundDark = active ? "orange.800" : "gray.500"
+        const iconBackgroundLight = active ? "blue.300" : "gray.400"
+        const display =
+          !showAnswer || id === selectedAnswer ? "inline-flex" : "none"
+        return (
+          <Button
```

Comment:
Shouldn't these be radio buttons?

Check this implementation example https://chakra-ui.com/docs/components/radio/usage#custom-radio-buttons